course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala

```diff
@@ -96,14 +100,19 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
     private def enrolUser(request: Request, courseId: String, batchId: String, userId: String): Unit = {
         val batchData: CourseBatch = courseBatchDao.readById(courseId, batchId, request.getRequestContext)
         val enrolmentData: UserCourses = userCoursesDao.read(request.getRequestContext, userId, courseId, batchId)
-        validateEnrolment(batchData, enrolmentData, true)
+        val id:util.List[String]=List(userId)
+        val ListOfUserDataForCourse:CourseUser=courseUserDao.read(request.getRequestContext,courseId,id)
+        val ListOfUserDataForBatch:BatchUser=batchUserDao.read(request.getRequestContext,batchId,id)
+
+        validateEnrolment(batchData, enrolmentData,true)
         val data: util.Map[String, AnyRef] = createUserEnrolmentMap(userId, courseId, batchId, enrolmentData, request.getContext.getOrDefault(JsonKey.REQUEST_ID, "").asInstanceOf[String])
         upsertEnrollment(userId, courseId, batchId, data, (null == enrolmentData), request.getRequestContext)
         logger.info(request.getRequestContext, "CourseEnrolmentActor :: enroll :: Deleting redis for key " + getCacheKey(userId))
         cacheUtil.delete(getCacheKey(userId))
         sender().tell(successResponse(), self)
         generateTelemetryAudit(userId, courseId, batchId, data, "enrol", JsonKey.CREATE, request.getContext)
-        notifyUser(userId, batchData, JsonKey.ADD)
+        notifyUser(userId, batchData,JsonKey.ADD)
+        validateUserCourseBAtchData(ListOfUserDataForCourse,ListOfUserDataForBatch,isEnrol=true)
```

Comment:
Lets discuss this point as I think it will not be needed

Reply:
done